src/app/account-settings/components/dialog-actions-duplicated-two-factor-auth/dialog-actions-duplicated-two-factor-auth.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component, Inject, OnInit } from '@angular/core'
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog'
+import { OauthService } from 'src/app/core/oauth/oauth.service'
+import { DuplicateRemoveEndpoint } from 'src/app/types/account-actions-duplicated'
+
+import { DialogActionsDuplicatedMergedConfirmedComponent } from '../dialog-actions-duplicated-merged-confirmed/dialog-actions-duplicated-merged-confirmed.component'
+
+@Component({
+  selector: 'app-dialog-actions-duplicated-two-factor-auth',
+  templateUrl: './dialog-actions-duplicated-two-factor-auth.component.html',
+  styleUrls: ['./dialog-actions-duplicated-two-factor-auth.component.scss'],
+})
+export class DialogActionsDuplicatedTwoFactorAuthComponent implements OnInit {
+  showBadVerificationCode: boolean
+  showBadRecoveryCode: boolean
+  loading: boolean
+  constructor(
+    private _oauth: OauthService,
+    private matRef: MatDialogRef<DialogActionsDuplicatedMergedConfirmedComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: DuplicateRemoveEndpoint
+  ) {}
+
+  ngOnInit(): void {}
+  authenticate(code: { verificationCode?: string; recoveryCode?: string }) {
+    this.showBadVerificationCode = false
+    this.showBadRecoveryCode = false
+    this.loading = true
+    this._oauth
+      .submitCodeForAnotherAccount({
+        orcid: this.data.deprecatingOrcid,
+        verificationCode: code.verificationCode || null,
+        recoveryCode: code.recoveryCode || null,
+        errors: [],
+        redirectUrl: null,
+      })
+      .subscribe((res) => {
+        this.loading = false
+        if (res.errors && res.errors.length > 0) {
+          if (res.errors[0].includes('verification')) {
```

Comment:
We should not use the `res.errors[0].includes('verification')` to check the 2FA response errors types since this will only work in English.
This is an issue we need to solve on the backend, by sending back an error code and not a string, so I will create a separate card to solve this on all the 2FA places.